homeassistant/components/bms_ble/__init__.py

```diff
@@ -0,0 +1,154 @@
+    )
+
+    if ble_device is None:
+        LOGGER.debug("Failed to discover device %s via Bluetooth", entry.unique_id)
+        raise ConfigEntryNotReady(
+            translation_domain=DOMAIN,
+            translation_key="device_not_found",
+            translation_placeholders={
+                "MAC": entry.unique_id,
+            },
+        )
+
+    plugin: ModuleType = await async_import_module(hass, entry.data["type"])
+    coordinator = BTBmsCoordinator(hass, ble_device, plugin.BMS(ble_device), entry)
+
+    # Query the device the first time, initialise coordinator.data
+    await coordinator.async_config_entry_first_refresh()
+
+    # Insert the coordinator in the global registry
+    hass.data.setdefault(DOMAIN, {})
+    entry.runtime_data = coordinator
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: BTBmsConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok: Final[bool] = await hass.config_entries.async_unload_platforms(
+        entry, PLATFORMS
+    )
+    LOGGER.debug("Unloaded config entry: %s, ok? %s!", entry.unique_id, str(unload_ok))
+
+    return unload_ok
+
+
+async def async_migrate_entry(
+    hass: HomeAssistant, config_entry: BTBmsConfigEntry
+) -> bool:
+    """Migrate old entry."""
+    _latest_version: Final[int] = 2
+
+    if config_entry.version > _latest_version:
+        # This means the user has downgraded from a future version
+        LOGGER.debug("Cannot downgrade from version %s", config_entry.version)
+        return False
+
+    LOGGER.debug(
+        "Migrating from version %s.%s", config_entry.version, config_entry.minor_version
+    )
+
+    if config_entry.version == 0:
+        bms_type: str = str(config_entry.data["type"])
+        new: dict[str, str] = {}
+        if bms_type == "OGTBms":
+            new = {"type": "custom_components.bms_ble.plugins.ogt_bms"}
+        elif bms_type == "DalyBms":
+            new = {"type": "custom_components.bms_ble.plugins.daly_bms"}
+        else:
+            LOGGER.debug("Entry: %s", config_entry.data)
+            LOGGER.error(
+                "Migration from version %s.%s failed",
+                config_entry.version,
+                config_entry.minor_version,
+            )
+            return False
+
+        hass.config_entries.async_update_entry(
+            config_entry, data=new, minor_version=0, version=1
+        )
+
+    if config_entry.version == 1:
+        bms_type = str(config_entry.data["type"])
+        new = {"type": f"aiobmsble.bms.{bms_type.rsplit('.', 1)[-1]}"}
+        hass.config_entries.async_update_entry(
+            config_entry, data=new, minor_version=0, version=2
+        )
+
+    LOGGER.debug(
+        "Migration to version %s.%s successful",
+        config_entry.version,
+        config_entry.minor_version,
+    )
+
+    return True
+
+
+def migrate_sensor_entities(
```

Comment:
The new integration should not have migrations.

Reply:
Removed. Thanks! Fixed in https://github.com/home-assistant/core/pull/159556/commits/b7c2e4f498de5d93abacd2043f38b40cb2760ea3